WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/MigrationDependencyContainer.swift

```diff
@@ -0,0 +1,46 @@
+class MigrationDependencyContainer {
+
+    let migrationCoordinator = MigrationFlowCoordinator()
+
+    func makeInitialViewController() -> UIViewController {
+        MigrationNavigationController(coordinator: migrationCoordinator,
+                                      migrationStack: [.welcome: makeWelcomeViewController(),
```

Comment:
I'm not opposed to this approach of having a dictionary mapping `MigrationStep` to a `UIViewController` but I just want to point out that these view controllers will stay in memory even when they're dismissed ( not visible ).

Perhaps we can optimize this by injecting a closure like this:

```swift
let closure = { [weak self] (step: MigrationStep) -> UIViewController? in 
     switch step {
           case .welcome: return self?. makeWelcomeViewController()
           ...
     }
}
```

Reply:
I like this idea: I added a factory that does exactly that, and we don't retain any unused vc in memory!